libvast/src/system/index.cpp

```diff
@@ -932,23 +934,11 @@ index(index_actor::stateful_pointer<index_state> self,
       auto client = caf::actor_cast<receiver_actor<atom::done>>(sender);
       // Sanity checks.
       if (!sender) {
-        VAST_ERROR("{} ignores an anonymous query", *self);
+        VAST_WARN("{} ignores query {} from anonymous sender", *self, query_id);
         return {};
       }
-      // A zero as second argument means the client drops further results.
       if (num_partitions == 0) {
-        VAST_DEBUG("{} drops remaining results for query id {}", *self,
-                   query_id);
-        if (auto it = self->state.monitored_queries.find(sender->address());
-            it != self->state.monitored_queries.end()) {
-          auto& [_, ids] = *it;
-          ids.erase(query_id);
-          if (ids.empty()) {
-            self->demonitor(sender->address());
-            self->state.monitored_queries.erase(it);
-          }
-        }
-        self->state.pending.erase(query_id);
+        VAST_WARN("{} ignores query {} for zero partitions", *self, query_id);
```

Comment:
This edit makes it impossible to "cancel" a query for a requester that wants to continue with other tasks. While we don't have the use-case right now we should not make such interface changes in a haste.

Reply:
We discussed this and came to the conclusion that we can live without this mechanism for now because it isn't used in any index client.

Reply:
We discussed this in a Slack call, and decided to keep this as-is in the PR. We do not have a use case for early cancellation without quitting the receiving actor, and if we want to add such an API back it is really simple to do so.